src/cli/java/org/commcare/util/cli/ApplicationHost.java

```diff
@@ -319,15 +320,20 @@ private boolean loopSession() throws IOException {
                     // which ultimately updates the session, so getNextScreen will move onto the form list,
                     // skipping the entity detail. To avoid this, flag that we want to force a redraw in this case.
                     boolean waitForCaseDetail = false;
-                    if (screen instanceof EntityScreen) {
-                        boolean isAction = input.startsWith("action "); // Don't wait for case detail if action
-                        EntityScreen eScreen = (EntityScreen)screen;
-                        if (!isAction && eScreen.getCurrentScreen() instanceof EntityListSubscreen) {
-                            waitForCaseDetail = true;
+                    if (screen instanceof MultiSelectEntityScreen) {
+                        String[] selectedValues = input.split(",");

```

Comment:
The semantics on `input` doesn't seem entirely clear to me here. I think we want to support 2 kinds of input `use_selected_values` or a guid to `MultiSelectEntityScreen` based on [this request lifecycle](https://github.com/dimagi/formplayer/wiki/Multi-Select-Case-Lists#http-lifecycle-bw-web-apps-and-formplayer). One idea is to take the selected values as - `use_selected_values val1 val2` while guid would just be a singular value like normal Entity Screen `guid_value` .  We should do input validation here to make sure input is one of these two types and throw an error in case of a mismatch, and parse the input so that we make either of these 2 calls depending on the input type  `screen.handleInputAndUpdateSession(mSession, USE_SELECTED_VALUES, false, selectedValues);` or for guid as input `screen.handleInputAndUpdateSession(mSession, input, false, null);`

Reply:
I don't think that level of complexity is necessary in the CLI since we aren't rebuilding the session on each navigation request. I don't think we should ever have to use the the GUID in the CLI and any input would always be the selected values.

Reply:
Is that what the CLI is for? I don't think about it like that. If that is the goal then we should really also be rebuilding the session on every 'request' and probably push a lot of the FP code down into the CLI.

I always thought the CLI was for testing apps.

Reply:
Ahh Interesting, looks like I probably am incorrect in my assumption here in that case. Makes sense to not use `guid` ever for CLI input.